src/app/dialog/dialog.service.ts

```diff
@@ -24,18 +24,29 @@ import {LinkTypeModel} from '../core/store/link-types/link-type.model';
 import {OrganizationModel} from '../core/store/organizations/organization.model';
 import {ProjectModel} from '../core/store/projects/project.model';
 import {DialogPath} from './dialog-path';
+import {environment} from '../../environments/environment';
+import {VideosAction} from '../core/store/videos/videos.action';
+import {getAllVideos} from '../core/store/videos/videos.data';
+import {AppState} from '../core/store/app.state';
+import {Store} from '@ngrx/store';
 
 /**
  * If callback is provided in any of the open*() methods, the calling component is responsible for closing the dialog
  * inside callback. If it was not implemented this way, the default redirect for closing the dialog would collide with
  * custom redirects made by a calling site inside the callback.
  */
-@Injectable()
+@Injectable({
+  providedIn: 'root',
+})
 export class DialogService {
   public callback: any;
   private open: boolean;
 
-  public constructor(private router: Router) {}
+  public constructor(private router: Router, private store: Store<AppState>) {
+    if (environment.videoKey) {
+      this.store.dispatch(new VideosAction.LoadVideos({videos: getAllVideos(), apiKey: environment.videoKey}));
```

Comment:
Doesn't it load videos twice? Once right before the redirect to Auth0 and again after coming back from it?

I don't think it belongs to this constructor or `DialogService` at all. Probably the best solution would be some `VideosGuard` on root view route or elsewhere.